core/domain/rights_manager.py

```diff
@@ -173,6 +211,8 @@ def update_activity_first_published_msec(
             since the Epoch.
     """
     activity_rights = _get_activity_rights(activity_type, activity_id)
+    # Ruling out the possibility of None for mypy type checking.
```

Comment:
What guarantees that activity_rights exist if this function is called?

(Checking because, for every such case, we actually need to make a decision whether to do error-handling or make an assertion. This is a general thing that's worth keeping in mind going forward.)

Reply:
In line 218, we are accessing the `first_published_msec` attribute without checking if `activity_rights` is none or not.

So, that's why I added the assertion here.

Reply:
See above, this is not sufficient justification.

Reply:
This function is never used in the codebase ( apart from test cases but in test cases, we are also not checking the none case of `activity_rights` ). So, shall I raise an expectation here!!

Reply:
If it is never used in the codebase, then you can delete it -- it's dead code.

Reply:
Removed!